frontend/src/app/forms/steps/basic-info/basic-info.component.ts

```diff
@@ -0,0 +1,71 @@
+/*
+@Author
+Bharathwaj Ravi
+
+Add modifiers under @Modifiers
+@Modifiers
+
+@Purpose
+  - This component renders the basic information step view and it's actions.
+*/
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+
+import { FormGroup } from './../../../common/forms/forms';
+import { BasicInfo } from './../../../interfaces/basic-info';
+
+@Component({
+  selector: 'form-basic-info',
+  templateUrl: './basic-info.component.html',
+  styleUrls: ['./basic-info.component.scss'],
+})
+export class BasicInfoComponent implements OnInit {
+  /*
+  @Usage It holds the current menu object from parent
+  */
+  @Input() menuObject: {
+    id: number;
+    displayName: string;
+    isCompleted: boolean;
+    isActive: boolean;
+    selecteor: string;
+  };
+
+  /*
+  @Usage It holds typesafe form group property fields.
+  */
+  @Input() basicInfoObject: FormGroup<BasicInfo>;
+
+  /*
+  @Usage It emits the progress callback on this step.
+  */
+  @Output() pageProgressStatusCallBack = new EventEmitter();
+
+  /*
+  @Usage It holds the array objects of drop down menu.
+  */
+  salutationsArray = [
+    { value: 'mr', displayName: 'Mr' },
+    { value: 'mrs', displayName: 'Mrs' },
+    { value: 'ms', displayName: 'Ms' },
+  ];
+
+  constructor() {}
+
+  ngOnInit(): void {}
+
+  /*
+  @Usage It handles the on change event for the fields and send emit call back event to parent.
+  */
+  onValChange(): void {
```

Comment:
In the HTML, you write `(ionChange)="onValChange()"` pretty often. I think it would be better to do:
``` TS
ngOnInit(): void {
  this.basicInfoObj.valueChanges.subscribe(() => {
      // Your logic here
  });
}

```
in your TS file instead. (So: Subscribe to the FormGroup's valueChanges and then react to the changes here.)

Reply:
Ya I am learning to code efficiently with Observers. I am not yet clear with it. In the next week (ionChange) will be converted as observer.